habitat/robots/static_manipulator.py

```diff
@@ -0,0 +1,458 @@
+        )
+
+    @arm_joint_pos.setter
+    def arm_joint_pos(self, ctrl: List[float]):
+        """Kinematically sets the arm joints and sets the motors to target."""
+        self._validate_arm_ctrl_input(ctrl)
+
+        joint_positions = self.sim_obj.joint_positions
+
+        for i, jidx in enumerate(self.params.arm_joints):
+            self._set_motor_pos(jidx, ctrl[i])
+            joint_positions[self.joint_pos_indices[jidx]] = ctrl[i]
+        self.sim_obj.joint_positions = joint_positions
+
+    def _validate_arm_ctrl_input(self, ctrl: List[float]):
+        """
+        Raises an exception if the control input is NaN or does not match the
+        joint dimensions.
+        """
+        if len(ctrl) != len(self.params.arm_joints):
+            raise ValueError(
+                "Control dimension does not match joint dimension"
+            )
+        if np.any(np.isnan(ctrl)):
+            raise ValueError("Control is NaN")
+
+    def set_fixed_arm_joint_pos(self, fix_arm_joint_pos):
+        """
+        Will fix the arm to a desired position at every internal timestep. Can
+        be used for kinematic arm control.
+        """
+        self._validate_arm_ctrl_input(fix_arm_joint_pos)
+        self._fix_joint_values = fix_arm_joint_pos
+        self.arm_joint_pos = fix_arm_joint_pos
+
+    @property
+    def arm_velocity(self) -> np.ndarray:
+        """Get the velocity of the arm joints."""
+
+        # deref self vars to cut access in half
+        joint_dof_indices = self.joint_dof_indices
+        arm_joints = self.params.arm_joints
+        sim_obj_joint_vel = self.sim_obj.joint_velocities
+
+        arm_dof_indices = (joint_dof_indices[x] for x in arm_joints)
+        return np.array(
+            [sim_obj_joint_vel[i] for i in arm_dof_indices],
+            dtype=np.float32,
+        )
+
+    @property
+    def arm_motor_pos(self) -> np.ndarray:
+        """Get the current target of the arm joints motors."""
+        motor_targets = np.zeros(len(self.params.arm_init_params))
+        for i, jidx in enumerate(self.params.arm_joints):
+            motor_targets[i] = self._get_motor_pos(jidx)
+        return motor_targets
+
+    @arm_motor_pos.setter
+    def arm_motor_pos(self, ctrl: List[float]) -> None:
+        """Set the desired target of the arm joint motors."""
+        self._validate_arm_ctrl_input(ctrl)
+
+        for i, jidx in enumerate(self.params.arm_joints):
+            self._set_motor_pos(jidx, ctrl[i])
+
+    def clip_ee_to_workspace(self, pos: np.ndarray) -> np.ndarray:
+        """Clips a 3D end-effector position within region the robot can reach."""
+        return np.clip(
+            pos,
+            self.params.ee_constraint[:, 0],
+            self.params.ee_constraint[:, 1],
+        )
+
+    @property
+    def arm_motor_forces(self) -> np.ndarray:
+        """Get the current torques on the arm joint motors"""
+        return np.array(self.sim_obj.joint_forces)
```

Comment:
As a more meta point, we may want to update the sim python bindings to just return numpy arrays in the future.